### PR TITLE
fix(ngcc): Esm5ReflectionHost.getDeclarationOfIdentifier should handle aliased inner declarations

### PR DESCRIPTION
**Pull request by petebacondarwin**
In ES5 modules, the class declarations consist of an IIFE with inner
and outer declarations that represent the class. The `EsmReflectionHost`
has logic to ensure that `getDeclarationOfIdentifier()` always returns the
outer declaration.

Before this commit, if an identifier referred to an alias of the inner
declaration, then `getDeclarationOfIdentifier()` was failing to find
the outer declaration - instead returning the inner declaration.

Now the identifier is correctly resolved up to the outer declaration
as expected.

This should fix some of the failing 3rd party packages discussed in
https://github.com/angular/ngcc-validation/issues/57.

### CODE REVIEW

File: packages/compiler-cli/ngcc/src/host/esm5_host.ts [hunk 1]
```diff
@@ -162,12 +162,25 @@ export class Esm5ReflectionHost extends Esm2015ReflectionHost {
...
+    const superDeclaration = super.getDeclarationOfIdentifier(id);
+
+    if (superDeclaration === null || superDeclaration.node === null) {
+      return null;
+    }
+
     // Get the identifier for the outer class node (if any).
-    const outerClassNode = getClassDeclarationFromInnerFunctionDeclaration(id.parent);
-    const declaration = super.getDeclarationOfIdentifier(outerClassNode ? outerClassNode.name : id);
+    const outerClassNode = ts.isFunctionDeclaration(superDeclaration.node) ?
```
Comment (gkalpak) [inline packages/compiler-cli/ngcc/src/host/esm5_host.ts]: This could be simplified to `const outerClassNode = getClassDeclarationFromInnerFunctionDeclaration(superDeclaration.node)`, since `ts.isFunctionDeclaration()` is the first check in `getClassDeclarationFromInnerFunctionDeclaration()`.

File: packages/compiler-cli/ngcc/src/host/esm5_host.ts [hunk 2]
```diff
@@ -162,12 +162,25 @@ export class Esm5ReflectionHost extends Esm2015ReflectionHost {
...
+        getClassDeclarationFromInnerFunctionDeclaration(superDeclaration.node) :
+        null;
+    const declaration = outerClassNode !== null ?
+        super.getDeclarationOfIdentifier(outerClassNode.name) :
+        superDeclaration;
+
+    if (!declaration || declaration.node === null) {
+      return null;
```
Comment (gkalpak) [inline packages/compiler-cli/ngcc/src/host/esm5_host.ts]: Previously, if `declaration` was non-null but `declaration.node` was null, we would return `declaration` (instead of `null`). Is this change intentional?

**Review by gkalpak — APPROVED** [review]
Nice catch :tada:

File: packages/compiler-cli/ngcc/src/host/esm5_host.ts (hunk 2 above)
Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/src/host/esm5_host.ts]: Non-intentional - I'll change it back, although I am not sure it will make much difference. It certainly doesn't affect the tests.
